src/components/StateBadge.vue

```diff
@@ -18,7 +18,7 @@
     dismissible?: boolean,
   }>()
 
-  const type = computed(() => mapStateNameToStateType(props.state?.name).type)
+  const type = computed(() => props.state?.type ?? mapStateNameToStateType(props.state?.name).type)
```

Comment:
In the example flow in the PR description, `props.state.type === 'completed'` already but the state has a custom name so before, it wouldn't map to any type.